rasa_core/validator.py

```diff
@@ -0,0 +1,394 @@
+)
+
+parser.add_argument(
+    '--no_validate_domain', action='store_true', default=False,
+    help='Skips validations to domain'
+)
+
+
+class Validator:
+    domain = ''
+    intents = []
+    stories = []
+    valid_intents = []
+    valid_utters = []
+
+    def __init__(self, domain="domain.yml", intents="data/nlu.md",
+                 stories="data/stories.md"):
+
+        # Saving domain file
+        if os.path.exists(domain):
+            self.domain = domain
+        else:
+            logger.error("The domain file was not found")
+
+        # Saving intents files
+        if os.path.isfile(intents) and os.path.exists(intents):
+            self.intents.append(intents)
+
+        elif os.path.isdir(intents):
+            if not intents.endswith('/'):
+                intents += '/'
+
+            intent_files = [f for f in listdir(intents)
+                            if isfile(join(intents, f))]
+            for file in intent_files:
+                self.intents.append(intents + file)
+
+        else:
+            logger.error("The intents file was not found")
+
+        # Saving stories files
+        if os.path.isfile(stories) and os.path.exists(stories):
+            self.stories.append(stories)
+
+        elif os.path.isdir(stories):
+            if not stories.endswith('/'):
+                stories += '/'
+
+            stories_files = [f for f in listdir(stories)
+                             if isfile(join(stories, f))]
+            for file in stories_files:
+                self.stories.append(stories + file)
+
+        else:
+            logger.error("The stories file was not found")
+
+    def verify_domain(self, warnings):
+        if self.domain != '':
+            schema = """
+            type: object
+            """
+            file = open(self.domain, 'r')
+            domain_file = file.read()
+            file.close()
+            try:
+                validate(yaml.load(domain_file), yaml.load(schema))
+                if warnings:
+                    self._check_spaces_between_utters()
+                logger.info('Domain verified')
+            except Exception as e:
+                logger.error('There is an error in ' + self.domain +
+                             ' ' + str(e))
+        else:
+            logger.error('The domain could not be verified')
+
+    def _check_spaces_between_utters(self):
+        file = open(self.domain, 'r')
+        domain_lines = file.readlines()
+        file.close()
+        for line in domain_lines:
+            line_s = line.strip().split('_')
+            if len(line_s) >= 2 and line_s[0] == 'utter':
+                index = domain_lines.index(line) - 1
+
+                previous_line = domain_lines[index]
+
+                if previous_line != '\n' and previous_line != 'templates:\n':
+                    logger.warning('There should be a space between lines ' +
+                                   str(index+1) + ' and ' + str(index+2) +
```

Comment:
Use `"{}".format` in this example as well

Reply:
solved [here](https://github.com/lappis-unb/rasa_core/blob/24d89908dbbab1081b9195949fb6ff7da05561e9/rasa_core/validator.py#L154)